### ISSUE TITLE
Nullable union of classes error

### ISSUE THREAD

**Issue by lookyman** [CuyZ/Valinor#315]
```php
class Foo
{
    public function __construct(
        public string $one,
        public string $two,
    ) {
    }
}

class Bar
{
    public function __construct(
        public string $three,
        public string $four,
    ) {
    }
}

class Wtf
{
    public function __construct(
        public string $six,
        public Foo|Bar|null $five = null,
    ) {
    }
}

(new MapperBuilder())
    ->mapper()
    ->map(Wtf::class, Source::array([
        'five' => [
            'one' => 'aaa',
            'two' => 'bbb',
        ],
        'six' => 'ccc',
    ]));
```

Throws `Could not map type 'Wtf'. An error occurred at path five: Invalid value array{one: 'aaa', two: 'bbb'}.` If I make the property not nullable, it works fine.

**Comment by lookyman** [comment]
Preliminary investigation leads me [here](https://github.com/CuyZ/Valinor/blob/1577233ac9dce444af61bb482f5a744e38eaf739/src/Mapper/Tree/Builder/UnionNodeBuilder.php#L57), but don't think it's the only place where some change must happen. I'll try to look into it some more when I have time.

**Comment by romm** [comment]
Hi @lookyman thanks for the report! The issue has been fixed with #348 and will be part of the next release. 😉
